dace/transformation/dataflow/redundant_array.py

```diff
@@ -275,7 +282,10 @@ def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
                 for a in accesses:
                     subsets_intersect = False
                     for e in graph.out_edges(a):
-                        subset, _ = _validate_subsets(e, sdfg.arrays, src_name=a.data)
+                        try:
+                            subset, _ = _validate_subsets(e, sdfg.arrays, src_name=a.data)
+                        except (NotImplementedError, ValueError):
```

Comment:
I am not sure if we actually want to raise an exception for any of those errors, i.e., there may be an actual bug in the code, rather than just being unable to infer the subsets. It might be worth to at least print something in debug mode, so someone can investigate if there are too many messages printed.

Reply:
Fixed now @alexnick83